frame/treasury/src/lib.rs

```diff
@@ -316,6 +321,16 @@ pub mod pallet {
 		/// - The weight is overestimated if some approvals got missed.
 		/// # </weight>
 		fn on_initialize(n: T::BlockNumber) -> Weight {
+			let pot = Self::pot();
+			let deactivated = Inactive::<T, I>::get();
+			match pot.checked_sub(&deactivated) {
+				Some(x) if !x.is_zero() => T::Currency::deactivate(x),
```

Comment:
Should not we update `Inactive`, on deactivate, `Inactive` = `Inactive` + x, and on reactivate `Inactive` = `Inactive` - x